rholang/src/main/scala/coop/rchain/rholang/interpreter/ParBuilder.scala

```diff
@@ -0,0 +1,117 @@
+package coop.rchain.rholang.interpreter
+
+import java.io.{Reader, StringReader}
+
+import cats.effect.Sync
+import cats.implicits._
+import coop.rchain.models.Connective.ConnectiveInstance
+import coop.rchain.models.Par
+import coop.rchain.models.rholang.implicits.VectorPar
+import coop.rchain.models.rholang.sorter.Sortable
+import coop.rchain.rholang.interpreter.errors._
+import coop.rchain.rholang.syntax.rholang_mercury.{parser, Yylex}
+import coop.rchain.rholang.syntax.rholang_mercury.Absyn.Proc
+
+trait ParBuilder[F[_]] {
+  def buildNormalizedTerm(source: String): F[Par]
+
+  def buildNormalizedTerm(reader: Reader): F[Par]
+
+  def buildPar(proc: Proc): F[Par]
+}
+
+object ParBuilder {
```

Comment:
In the compiler theory is there a better name suitable for this, instead of ParBuilder?